repos/system_upgrade/el7toel8/actors/storagescanner/libraries/library.py

```diff
@@ -79,22 +80,32 @@ def _get_fstab_info(fstab_path):
     ''' Collect storage info from /etc/fstab file '''
     if _is_file_readable(fstab_path):
         with open(fstab_path, 'r') as fstab:
-            for entry in fstab:
+            keys = ['fs_spec', 'fs_file', 'fs_vfstype', 'fs_mntops', 'fs_freq', 'fs_passno']
+            for line, entry in enumerate(fstab, 1):
                 if entry.startswith('#'):
                     continue
 
+                if '#' in entry:
+                    # Ignore comments at end of lines.
+                    entry = entry.split('#')[0]
```

Comment:
This seems wrong, as from the documentation:
> Lines starting with '#' are comments.

Regarding the tests, fstab can even contain "#" comments in the end of line (even when this is not documented and it is not recommended to do). So in case this occurs, I would log warning or something. Keep in mind, that current parsing we are doing is not 100% right, as even the hashtag carachter can be used as part of value (e.g. name or password) and do not have to mean it is comment. So I would be careful here and rather than split, I would log error and require remove of the comment from the line or at least log warning.